packages/actor-bindings-aggregator-factory-average/lib/AverageAggregator.ts

```diff
@@ -33,8 +32,8 @@ export class AverageAggregator extends AggregateEvaluator implements IBindingsAg
       this.state = { sum, count: 1 };
     } else {
       const internalTerm = this.termToNumericOrError(term);
-      this.state.sum = <E.NumericLiteral> this.summer.apply([ this.state.sum, internalTerm ],
-        <ExpressionEvaluator> this.evaluator);
+      this.state.sum = <E.NumericLiteral> this.additionFunction.applyOnTerms([ this.state.sum, internalTerm ],
+        (<ExpressionEvaluator> this.evaluator).internalizedExpressionEvaluator);
```

Comment:
This feels a bit icky. Would it be feasible to avoid the cast? Perhaps we can just pass the IEvaluator interface?

Reply:
The more I think abut how to solve this, the less I feel like we should have an interface like `IExpressionEvaluator`. We need to access to context here, if I describe that context in an interface, we lose all generality the interface would bring? At that point, we just have the added complexity of an interface without any gains?

Reply:
Based on our last chat, I think bus-ifying this may solve this problem.

Reply:
Fixed by extracting an ActionConfig.